hivemind/optim/collaborative.py

```diff
@@ -344,38 +370,36 @@ def accumulated_grads(self) -> Iterator[torch.Tensor]:
         """local gradient accumulators"""
         if self.reuse_grad_buffers:
             yield from self._grad_buffers()
-        elif self._grads is None:
-            with torch.no_grad():
-                self._grads = [
-                    torch.zeros_like(grad, device=self.accumulate_grads_on) for grad in self._grad_buffers()
-                ]
-        yield from self._grads
```

Comment:
This wold actually an error with reuse_grad_buffers=True, but it worked because noone asked for more than len(grad_buffers) elements